Defining DWORD inside fakeit namespace causes issues due to pre-2017 MSVC bug

MSVC versions before 2017 (i.e. 2015 and older) have a bug where defining a `typedef` inside a namespace causes name lookup issues if a given translation unit uses that namespace inside an anonymous namespace. Try compiling the code below with any pre-2017 version :

```
namespace foobar { typedef unsigned long myulong; }
typedef unsigned long myulong;
namespace { using namespace foobar; }
void a(int b) { static_cast<myulong>(b); }
```

The compiler will throw following errors :
```
err.cpp(4): error C2872: 'myulong': ambiguous symbol
err.cpp(2): note: could be 'unsigned long myulong'
err.cpp(1): note: or       'foobar::myulong'
```

Analogous errors happen when including fakeit.h and trying to use `DWORD` in any of the translation unit's global functions later on. Here's a minimal example :

```
#include "fakeit.hpp"
#include <windows.h>
namespace { using namespace fakeit; }
void globalfn(int x) { static_cast<DWORD>(x); }
```

--- a/include/mockutils/mscpp/VirtualTable.hpp
+++ b/include/mockutils/mscpp/VirtualTable.hpp
@@ -9,7 +9,7 @@
 
 namespace fakeit {
 
-    typedef unsigned long DWORD;
+    typedef unsigned long dword_;
 
     struct TypeDescriptor {
         TypeDescriptor() :
@@ -22,7 +22,7 @@ namespace fakeit {
         }
 
 		char *ptrToVTable;
-        DWORD spare;
+        dword_ spare;
         char name[8];
     };
 
@@ -50,9 +50,9 @@ namespace fakeit {
         }
 
         const std::type_info *pTypeDescriptor; //type descriptor of the class
-        DWORD numContainedBases; //number of nested classes following in the Base Class Array
+        dword_ numContainedBases; //number of nested classes following in the Base Class Array
         struct PMD where;        //pointer-to-member displacement info
-        DWORD attributes;        //flags, usually 0
+        dword_ attributes;        //flags, usually 0
     };
 
     template<typename C, typename... baseclasses>
@@ -74,9 +74,9 @@ namespace fakeit {
             delete[] pBaseClassArray;
         }
 
-        DWORD signature;      //always zero?
-        DWORD attributes;     //bit 0 set = multiple inheritance, bit 1 set = virtual inheritance
-        DWORD numBaseClasses; //number of classes in pBaseClassArray
+        dword_ signature;      //always zero?
+        dword_ attributes;     //bit 0 set = multiple inheritance, bit 1 set = virtual inheritance
+        dword_ numBaseClasses; //number of classes in pBaseClassArray
         RTTIBaseClassDescriptor **pBaseClassArray;
 
         template<typename BaseType>
@@ -109,11 +109,11 @@ namespace fakeit {
 		{
 		}
 
-		DWORD signature; //always zero ?
-		DWORD offset;    //offset of this vtable in the complete class
-		DWORD cdOffset;  //constructor displacement offset
-		DWORD typeDescriptorOffset;
-		DWORD classDescriptorOffset;
+		dword_ signature; //always zero ?
+		dword_ offset;    //offset of this vtable in the complete class
+		dword_ cdOffset;  //constructor displacement offset
+		dword_ typeDescriptorOffset;
+		dword_ classDescriptorOffset;
 #else
 		RTTICompleteObjectLocator(const std::type_info &info) :
 			signature(0), offset(0), cdOffset(0),
@@ -125,9 +125,9 @@ namespace fakeit {
 			delete pClassDescriptor;
 		}
 
-		DWORD signature; //always zero ?
-		DWORD offset;    //offset of this vtable in the complete class
-		DWORD cdOffset;  //constructor displacement offset
+		dword_ signature; //always zero ?
+		dword_ offset;    //offset of this vtable in the complete class
+		dword_ cdOffset;  //constructor displacement offset
 		const std::type_info *pTypeDescriptor; //TypeDescriptor of the complete class
 		struct RTTIClassHierarchyDescriptor<C, baseclasses...> *pClassDescriptor; //describes inheritance hierarchy
 #endif

